distributed/utils_test.py

```diff
@@ -104,6 +104,22 @@ def is_debugging() -> bool:
 _offload_executor.submit(lambda: None).result()  # create thread during import
 
 
+class _Watch:
+    def __init__(self, duration: float):
+        self.duration = duration
```

Comment:
I think it would be better to store the deadline rather than the separate duration and started_at, a bit like the new `asyncio.timeout()`: https://github.com/python/cpython/blob/70690c7233aaad4431660a64daef81b5220e2ac1/Lib/asyncio/timeouts.py#L145

Reply:
FYI: https://github.com/dask/distributed/blob/2ab8cbd3b3f164a25665ee9195b876f9122a40f9/distributed/utils.py#L1779 has been added since.